ethereum/core/src/test/java/org/hyperledger/besu/ethereum/MainnetBlockValidatorTest.java

```diff
@@ -223,4 +223,48 @@ public void shouldReturnBadBlockBasedOnTheHash() {
         HeaderValidationMode.DETACHED_ONLY);
     assertThat(badBlockManager.getBadBlock(badBlock.getHash())).containsSame(badBlock);
   }
+
+  @Test
+  public void shouldNotAddBadBlockToBadBlockManagerWhenShouldNotRecordBadBlockFlagIsSetToTrue() {
```

Comment:
I found this confusing to read at first. Perhaps instead of "WhenShouldNotRecordBadBlockFlagIsSetToTrue" this could be "WhenShouldRecordBadBlockIsFalse" to closer match passing the false as the value for shouldRecordBadBlock.